Public/Src/Cache/ContentStore/Distributed/NuCache/EventStreaming/ContentLocationEventData.cs

```diff
@@ -97,13 +101,15 @@ public static ContentLocationEventData Deserialize(BuildXLReader reader, DateTim
                     return new TouchContentLocationEventData(sender, hashes, eventTimeUtc);
                 case EventKind.Reconcile:
                     return new ReconcileContentLocationEventData(sender, reader.ReadString());
+                case EventKind.UpdateMetadataEntry:
+                    return new UpdateMetadataEntryEventData(sender, reader);
                 default:
                     throw new ArgumentOutOfRangeException($"Unknown event kind '{kind}'.");
             }
         }
 
         /// <nodoc />
-        public void Serialize(BuildXLWriter writer)
+        public virtual void Serialize(BuildXLWriter writer)

```

Comment:
Minor, but curious why you decided to use another pattern here?

Before the full dispatch was happened here, and now it is virtual. I think it make sense to stay consistent unless there is a reason to follow different pattern in different cases.